### PR TITLE
pointer: Add helper funcs for time.Duration

### PR DESCRIPTION
**Pull request by ialidzhikov**
/kind feature

**What this PR does / why we need it**:
The PR adds helper funcs to the pointer pkg for working with `time.Duration`.

**Release note**:
```
NONE
```

### CODE REVIEW

**Review by MadhavJivrajani — COMMENTED** [review]
Could you please add unit tests?

**Comment by thockin** [comment]
All the other types have a "simple" func name, eg. `pointer.Float64` - can you do the same with `Duration`?  The longer `Float64Ptr` is a back-compat thing.

IOW - there's a pattern, is there a reason not to follow it?

**Comment by ialidzhikov** [comment]
> Could you please add unit tests?

Done.

> All the other types have a "simple" func name, eg. `pointer.Float64` - can you do the same with `Duration`? The longer `Float64Ptr` is a back-compat thing.
> 
> IOW - there's a pattern, is there a reason not to follow it?

Good point, I now changed from `DurationPtr` to `Duration`.

**Comment by ialidzhikov** [comment]
@MadhavJivrajani @thockin did you had a chance to check the recent changes?

**Comment by ialidzhikov** [comment]
Friendly ping @MadhavJivrajani @thockin

File: pointer/pointer_test.go [hunk 1]
```diff
@@ -372,3 +373,49 @@ func TestFloat64Equal(t *testing.T) {
...
+	if DurationEqual(nil, Duration(42)) {
+		t.Errorf("expected false (nil != val)")
+	}
+	if DurationEqual(Duration(42), nil) {
+		t.Errorf("expected false (val != nil)")
+	}
+	if DurationEqual(Duration(42), Duration(43)) {
+		t.Errorf("expected false (val != val)")
+	}
+}
```
Comment (MadhavJivrajani) [inline pointer/pointer_test.go:421]: These look good!
Would it be possible to add these tests in a tabular form?

To be clear, I mean something like:
```go
func TestDuration(t *testing.T) {
    cases := []struct{
        name string // if applicable, optional
        val time.Duration
        ptr *time.Duration
    } {
        {...},
    }
    for _, case := range cases {
        ...
    }
}
```

**Review by MadhavJivrajani — COMMENTED** [review]
Apologies for the late reply, suggested a minor change, thanks for raising the PR!

File: pointer/pointer_test.go (hunk 1 above)
Comment (ialidzhikov) [reply pointer/pointer_test.go:421]: These tests are "inspired" by the existing tests that all follow the same approach. For consistency reasons, I would rather keep it as it is. I guess you can open issue to rework all existing tests to table tests. WDYT?

Comment (MadhavJivrajani) [reply pointer/pointer_test.go:421]: Let's keep it as is :)

**Comment by MadhavJivrajani** [comment]
/assign @thockin 
in case you're around to take a final look and approve :)
Thanks!

**Review by thockin — COMMENTED** [review]
Thanks!

/lgtm
/approve

**Pull request merged**